File: E-You App/enviromental-app-develop_dragos/Android/app/src/main/java/com/example/greenlight/di/AppComponent.java
package com.example.greenlight.di;

import com.example.greenlight.EYouApp;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        MainActivityModule.class,
        AppModule.class,
        ExploreFragmentModule.class,
        SplashScreenModule.class,
        ProfileFragmentModule.class,
        SignUpFragmentModule.class,
        DatePickerFragmentModule.class,
        SignInFragmentModule.class,
        MapContainerFragmentModule.class,
        ApplicationActivityModule.class,
        GoogleMapFragmentModule.class,
        ReportActionFragmentModule.class,
        ReportDetailFragmentModule.class,
        SolveIncidentFragmentModule.class,
        RankingFragmentModule.class,
        InfoMarkerFragmentModule.class,
        AndroidSupportInjectionModule.class
})


public interface AppComponent {
    void inject(EYouApp instance);
}

File: LICENTA_GREENLIGHT/Android/app/src/main/java/com/ciurezu/gheorghe/dragos/greenlight/di/AppComponent.kt
package com.ciurezu.gheorghe.dragos.greenlight.di

import com.ciurezu.gheorghe.dragos.greenlight.GreenLightApp
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        AndroidInjectionModule::class,
        FragmentSignUpNameModule::class,
        FragmentSplashScreenModule::class,
        MainActivityModule::class,
        FragmentChooseSignInOrUpModule::class,
        FragmentIntroModule::class,
        FragmentMapScreenModule::class,
        FragmentSignInModule::class,
        FragmentRankingScreenModule::class,
        FragmentHomeScreenModule::class,
        GreenLightActivityModule::class,
        FragmentReportDashboardModule::class,
        FragmentReportIncidentModule::class,
        FragmentFAQScreenModule::class,
        FragmentAchievementScreenModule::class,
        FragmentShoppingBagModule::class,
        FragmentShoppingBuyModule::class,
        FragmentShoppingScreenModule::class,
        FragmentTrashScreenModule::class,
        FragmentSettingsScreenModule::class,
        FragmentInfoMarkerModule::class,
        FragmentSolveMarkerModule::class,
        FragmentCompanyScreenModule::class,
        FragmentSignUpIdentificationModule::class,
        AppModule::class
    ]
)
@Singleton
interface AppComponent {
    fun inject(application: GreenLightApp)
}